src/problems/MetalAdvectionProblem/test_sne.cpp

```diff
@@ -0,0 +1,540 @@
+};
+
+template <> void QuokkaSimulation<NewProblem>::setInitialConditionsOnGrid(quokka::grid const &grid_elem)
+{
+
+	amrex::GpuArray<amrex::Real, AMREX_SPACEDIM> dx = grid_elem.dx_;
+	amrex::GpuArray<amrex::Real, AMREX_SPACEDIM> prob_lo = grid_elem.prob_lo_;
+	const amrex::Box &indexRange = grid_elem.indexRange_;
+	const amrex::Array4<double> &state_cc = grid_elem.array_;
+
+	double vol = AMREX_D_TERM(dx[0], *dx[1], *dx[2]);
+
+	amrex::ParallelFor(indexRange, [=] AMREX_GPU_DEVICE(int i, int j, int k) {
+		amrex::Real const z = prob_lo[2] + (k + amrex::Real(0.5)) * dx[2];
+
+		// Calculate DM Potential
+		double prefac;
+		prefac = 2. * M_PI * Const_G * rho_dm * std::pow(R0_Gal, 2);
+		double Phidm = (prefac * std::log(1. + std::pow(z / R0_Gal, 2)));
+
+		// Calculate Stellar Disk Potential
+		double prefac2;
+		prefac2 = 2. * M_PI * Const_G * Sigma_star * z_star;
+		double Phist = prefac2 * (std::pow(1. + z * z / z_star / z_star, 0.5) - 1.);
+
+		// Calculate Gas Disk Potential
+
+		double Phigas;
+		// Interpolate to find the accurate g-value from array-- because linterp doesn't work on Setonix
+		// TODO - AV to find out why linterp doesn't work
+		size_t ii = 0;
+		double x_interp = std::abs(z);
+		while (ii < z_data.size() - 1 && x_interp > z_data[ii + 1]) {
+			ii++;
+		}
+
+		// Perform linear interpolation
+		const Real x1 = z_data[ii];
+		const Real x2 = z_data[ii + 1];
+		const Real y1 = logphi_data[ii];
+		const Real y2 = logphi_data[ii + 1];
+		amrex::Real phi_interp = (y1 + (y2 - y1) * (x_interp - x1) / (x2 - x1));
+		Phigas = std::pow(10., phi_interp);
+
+		double Phitot = Phist + Phidm + Phigas;
+
+		double rho, rho_disk, rho_halo;
+		rho_disk = rho01 * std::exp(-Phitot / std::pow(sigma1, 2.0));
+		rho_halo = rho02 * std::exp(-Phitot / std::pow(sigma2, 2.0)); // in g/cc
+		rho = (rho_disk + rho_halo);
+
+		double P = rho_disk * std::pow(sigma1, 2.0) + rho_halo * std::pow(sigma2, 2.0);
+
+		AMREX_ASSERT(!std::isnan(rho));
+
+		const auto gamma = HydroSystem<NewProblem>::gamma_;
+
+		state_cc(i, j, k, HydroSystem<NewProblem>::density_index) = rho;
+		state_cc(i, j, k, HydroSystem<NewProblem>::x1Momentum_index) = 0.0;
+		state_cc(i, j, k, HydroSystem<NewProblem>::x2Momentum_index) = 0.0;
+		state_cc(i, j, k, HydroSystem<NewProblem>::x3Momentum_index) = 0.0;
+		state_cc(i, j, k, HydroSystem<NewProblem>::internalEnergy_index) = P / (gamma - 1.);
+		state_cc(i, j, k, HydroSystem<NewProblem>::energy_index) = P / (gamma - 1.);
+		state_cc(i, j, k, Physics_Indices<NewProblem>::pscalarFirstIndex) = 1.e-5 / vol; // Injected tracer
+	});
+}
+
+void AddSupernova(amrex::MultiFab &mf, amrex::GpuArray<Real, AMREX_SPACEDIM> prob_lo, amrex::GpuArray<Real, AMREX_SPACEDIM> prob_hi,
+		  amrex::GpuArray<Real, AMREX_SPACEDIM> dx, SimulationData<NewProblem> const &userData, int level)
+{
+	// TODO for AV - ave (and restore) the RNG state in the metadata.yaml file
+	//  inject energy into cells with stochastic sampling
```

Comment:
I think it would be good to do this, so the simulations are reproducible when restarted. It should help debugging a lot.